### ISSUE TITLE
What's the runtime of this section? Can we do better?

### ISSUE THREAD

**Issue by yablonsky** [AmadoSuarez/training_problems#2]
https://github.com/AmadoSuarez/training_problems/blob/7a83e1c27683850a206a80eed1339f7663b167f6/valid_estring.py#L10-L11

**Comment by AmadoSuarez** [comment]
Yes, we can do better. The for-loop's running time is n by m.  By using the Counter datatype we can improve that.
